packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -950,4 +956,113 @@ router.post(
   })
 );
 
+/**
+ * POST /internal/patient/consolidated/query
+ *
+ * For each patient, get all consolidated queries that are older than 1 hour and are still processing, and:
+ * - update the status of the query to "failed"; and
+ * - start a new consolidated query with the same parameters.
+ *
+ * @param req.query.patientIds The patient IDs.
+ * @param req.query.minAgeInMinutes The minimum age in minutes for the queries to be processed (optional,
+ *        defaults to 60 minutes).
+ * @param req.query.dryRun Whether to simply return what would be done or actually execute the changes
+ *        and commands.
+ * @return The list of consolidated queries that where be executed (or would be executed, if dryRun is true).
+ */
+router.post(
+  "/consolidated/query",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const patientIds = getFromQueryAsArrayOrFail("patientIds", req);
+    const skipWebhooks = getFromQueryAsBoolean("skipWebhooks", req);
+    if (skipWebhooks === undefined) throw new BadRequestError("skipWebhooks is required");
+    const dryRun = getFromQueryAsBoolean("dryRun", req);
+    if (dryRun === undefined) throw new BadRequestError("dryRun is required");
+    const minAgeRaw = getFromQuery("minAgeInMinutes", req);
+    const minAgeInMinutes = minAgeRaw ? parseInt(minAgeRaw) : 60;
+
+    const patientsNotFound: string[] = [];
+    const patientsUpdated: string[] = [];
+    const patientsWithoutQueries: string[] = [];
+
+    const queryPromises: Promise<ConsolidatedQuery>[] = [];
+    const triggeredQueries: ConsolidatedQueryParams[] = [];
+    for (const patientId of patientIds) {
+      const { log } = out(`patientId ${patientId} minAge ${minAgeInMinutes} dryRun ${dryRun}`);
+      await executeOnDBTx(PatientModel.prototype, async transaction => {
+        const patient = await PatientModel.findOne({
+          where: { id: patientId },
+          transaction,
+        });
+        if (!patient) {
+          patientsNotFound.push(patientId);
+          log(`not found`);
+          return;
+        }
+        const consolidatedQueries = patient.data.consolidatedQueries ?? [];
+        const queriesToProcess = consolidatedQueries.filter(
+          query =>
+            query.status === "processing" &&
+            buildDayjs(query.startedAt).isBefore(
+              buildDayjs().subtract(minAgeInMinutes, "minutes").toISOString()
+            )
+        );
+        log(`queriesToProcess ${queriesToProcess.length}`);
+        if (queriesToProcess.length > 0) {
+          patientsUpdated.push(patientId);
+          queriesToProcess.forEach(query => {
+            query.status = "failed";
+          });
+          if (!dryRun) {
+            const data = {
+              ...patient.data,
+              consolidatedQueries,
+            };
+            patient.changed("data", true);
+            await patient.update({ data });
```

Comment:
It doesnt seem like you would be changing anything here. You are setting query.status to failed but that i dont think would ovveride consolidatedQueries variable.

Reply:
Rather it would update queriesToProcess const but thats not being used to override the data

Reply:
It's updated by reference :)